sn_client/src/chunk_operator.rs

```diff
@@ -0,0 +1,160 @@
+use bytes::Bytes;
+use self_encryption::MIN_ENCRYPTABLE_BYTES;
+use sn_protocol::storage::{Chunk, ChunkAddress};
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use xor_name::XorName;
+
+/// Tries to chunk the file, returning
+/// `(head_address, data_map_chunk, file_size, chunk_names)`
+/// and writes encrypted chunks to disk.
+#[derive(Debug, PartialEq)]
+pub struct ChunkOperator {
+    pub head_chunk_address: ChunkAddress,
+    pub data_map: Chunk,
+    pub file_size: u64,
+    pub chunk_vec: Vec<(XorName, PathBuf)>,
+}
```

Comment:
Is this the result of the `BuildChunkOperator`? If so, can we state that in the docs? I think the docs is misleading. Maybe `ChunkOperatorResult` would work here? We use the convention in our codebase.